dummy-data-product/src/dependencies/scraping/scraper.py

```diff
@@ -0,0 +1,157 @@
+        try:
+          li4.append(sector.contents[0].split(';')[2])
+        except IndexError:
+          li4.append(np.nan)
+
+      #scraping consultant source and budget
+      for i in document.find_all('div',{'class':'item-title'}):
+        for link in i.find_all('a'):
+          url = 'https://www.adb.org'+ link.get('href')
+          li7.append(url)
+
+          #scraping source if it's not a pdf file
+          if (requests.get(url).url[-3:] != 'pdf'):
+            r = requests.get(url)
+            re_link = requests.get(r.url)
+            scrp = re_link.text
+            document1 = BeautifulSoup(scrp, 'html.parser')
+
+            #scraping the country code and name if it's not a pdf file
+            for i in document1.find_all('span',{'id':'atAgency:mstCountry:0'}):
+              country_val = i.contents[0]
+              li5.append(country_val)
+              try:
+                code = pycountry.countries.search_fuzzy(country_val)[0].numeric
+                li6.append(code)
+              except:
+                li6.append(np.nan)
+                
+            #scarping source
+            if (document1.find_all('span',{'id':'mcConsultantSource'})):
+              for i in document1.find_all('span',{'id':'mcConsultantSource'}):
+                source = i.contents[0].text
+                li8.append(source)
+            else:
+              li8.append(np.nan)
+              
+          #scraping budget, if the file isn't PDF. 
+            if (document1.find_all('span',{'id':'mstBudgetAmount'})):
+              for j in document1.find_all('span',{'id':'mstBudgetAmount'}):
+                budget = j.contents[0]
+                li9.append(budget)
+            else:
+                li9.append(np.nan)
+          else: #if the file is a pdf file, asigning it a NA string.
+            li8.append(np.nan)
+            li9.append(np.nan)
+            li5.append(np.nan)
+            li6.append(np.nan)
+            
+      data = pd.DataFrame({'Name':li})
+      data['Status'] = li2
+      data['Date'] = li3
+      data['Sector'] = li4
+      data['Country'] = li5
+      data['country_code'] = li6
+      
+      #Adding Map Co-ordinates
+      #Since, map co-ordinates aren't given, puttin [] in place.
+      data['map_cordinates'] = '[]'
+      data['project_or_tender'] = 'T'   #scraping only tender data for this project
+      data['url'] = li7
+      data['source'] = li8
+      data['budget'] = li9
+      
+    return data
+
+  def clean(self):
+
+    data_to_clean = self.scrape()
+    clean_data = cleaning.cleaning(data_to_clean).clean()
+
+    return (clean_data)    
+
+  def standards(self):
+    
+    data_to_standardize = self.clean()
+    standardize_data = standardizer.standardize(data_to_standardize).standard()
+    return (standardize_data)
+  
+  def run(self):
+    output = self.standards()
+    return (output.to_csv('adb.csv', index = False))
+
+if __name__ == "__main__":
+  config = {}

```

Comment:
Why is this config empty?

How are you referring to the data you have scraped during the test run?
And how are you updating it?